### DISCUSSION TITLE
Question: possible to add, remove, or merge layers in a texture array

### DISCUSSION THREAD

**Discussion by theahura** [KhronosGroup/KTX-Software#694] [Q&A] [answered]
Hey folks, quick question and apologies if this isn't the right place.

I've been working with ktx2 files to compress texture atlases, primarily using basisu compression. I was interested in switching to texture arrays because I was thinking it may be possible to stitch together many previously compressed files into one texture array.

After reading through the ktx2 spec, I couldn't find anything that says this isn't possible; at the same time, I'm equally not confident that it is. 

Tldr: is it possible to add new layers to an already existing ktx2 texture array? Is it possible to merge two such texture arrays?

Thanks in advance!

**Comment by MarkCallow** [comment, marked as answer]
You can certainly compress array textures with basisu compression.

If you are using BasisLZ/ETC1S compression then you need to compress all the layers together because of the global codebook which also means that you will have better results than compressing the layers separately. Therefore you need to create an array texture from your original source images then compress that.

For all other formats you could write a tool to read a bunch of files (all having the same format) and create a new array texture with each layer being the contents of one of those other textures. libktx provides the features needed. No such tool exists at present.
